### PR TITLE
Fixed Compiled Issues in `example/nodejs`(escrow example)

### PR DESCRIPTION
**Pull request by nhenin**
## Summary by CodeRabbit

- **Refactor**
  - Streamlined address handling across various components, enhancing the consistency and reliability of address-related operations.

- **Bug Fixes**
  - Addressed issues with address validation and representation to ensure accurate processing and display of address information.

- **Documentation**
  - Updated documentation to reflect changes in address handling and validation mechanisms.

- **Chores**
  - Removed deprecated address-related code to maintain a clean and efficient codebase.

### CODE REVIEW

File: packages/runtime/core/src/address.ts [hunk 1]
```diff
@@ -2,19 +2,27 @@ import * as t from "io-ts/lib/index.js";
...
+export const AddressBech32Guard = t.brand(
+  t.string,
+  (s): s is t.Branded<string, AddressBech32Brand> => true,
+  "AddressBech32"
+);
+
+export type AddressBech32 = t.TypeOf<typeof AddressBech32Guard>;
+
+export const addressBech32 = (s: string) =>
+  unsafeEither(AddressBech32Guard.decode(s));
```
Comment (hrajchert) [inline packages/runtime/core/src/address.ts:20]: this is not needed (and thus the dependency to adapter)
This screenshot is from the old code (that still uses t.brand) and it works fine

Screenshot 2023-12-13 at 11 01 56

Comment (hrajchert) [reply packages/runtime/core/src/address.ts:20]: in there, x is a Branded type

File: packages/runtime/core/src/tsconfig.json [hunk 2]
```diff
@@ -1,6 +1,14 @@
 {
   "extends": "../../../../tsconfig-base.json",
   "compilerOptions": {
-    "outDir": "../dist/esm"
-  }
+    "outDir": "../dist/esm",
+    "paths": {
+      "@marlowe.io/adapter/*": ["../../../../adapter/src/*"],
```
Comment (hrajchert) [inline packages/runtime/core/src/tsconfig.json:6]: the adapter part should not be needed with what I added

**Review by hrajchert — CHANGES_REQUESTED** [review]
minor changes

File: packages/runtime/core/src/tsconfig.json (hunk 2 above)
Comment (nhenin) [reply packages/runtime/core/src/tsconfig.json:6]: since I have kept `addressBech32` function, this fix is still relevant I believe

**Review by hrajchert — APPROVED** [review]

**Pull request merged**